File: Machine_Learning/Regression/Polynomial_Linear_Regression/Plr.py
# -*- coding: utf-8 -*-
"""
Created on Wed Dec  4 19:30:06 2019

@author: rhari
"""

import numpy as np
import matplotlib.pyplot as plt
import pandas as pd

dataset=pd.read_csv('Position_Salaries.csv')
X=dataset.iloc[:,1:2].values
y=dataset.iloc[:,2].values

"""from sklearn.model_selection import train_test_split
X-train, X_test, y_train, y-test=train-test-split(X,y,test_size=0.2, random_state=0)"""

from sklearn.linear_model import LinearRegression
lin_reg=LinearRegression()
lin_reg.fit(X,y)

from sklearn.preprocessing import PolynomialFeatures
poly_reg=PolynomialFeatures(degree=4)
X_poly=poly_reg.fit_transform(X)
lin_reg_2=LinearRegression()
lin_reg_2.fit(X_poly, y)

#Visiualising the Linear Regression Model
plt.scatter(X,y, color='red')
plt.plot(X, lin_reg.predict(X), color='blue')
plt.title('Truth or Bluff(Linear Regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

#Visualising the Polynomial Regression Model
plt.scatter(X, y, color='red')
plt.plot(X, lin_reg_2.predict(poly_reg.fit_transform(X)), color='blue')
plt.title('Truth or Bluff(Polynonial Regression)')
plt.xlabel('Position level')
plt.ylabel('Salary')
plt.show()

#Predicting a new result with Linear Regression
lin_reg.predict(np.reshape(6.5,(1,1)))

#Predicting a new result with Polynomial Regression
lin_reg_2.predict(poly_reg.fit_transform(np.reshape(6.5,(1,1))))

File: Machine_Learning/Regression/Polynomial_Linear_Regression/Plr.r
#Importing the dataset
dataset=read.csv("Position_Salaries.csv")
dataset=dataset[, 2:3]

#Training and test set split
# library(caTools)
# set.seed(123)
# split=sample.split(dataset$Salary, SplitRatio = 0.8)
# training_set=subset(dataset, split==TRUE)
# test_set=subset(dataset, split==FALSE)

#Linear Regression
lin_reg=lm(formula = Salary~., data= dataset)

#Polynomial Regression
dataset$Level2=dataset$Level^2
dataset$Level3=dataset$Level^3
dataset$Level4=dataset$Level^4
dataset$Level5=dataset$Level^5
poly_reg=lm(formula = Salary~., data=dataset)

#Visualising
library(ggplot2)
ggplot()+
  geom_point(aes(x=dataset$Level, y=dataset$Salary), colour="red")+
  geom_line(aes(x=dataset$Level, y=predict(poly_reg, newdata = dataset)), colour="blue")
